File: src/main/java/tv/comnata/comnata/configs/AppConfig.java
package tv.comnata.comnata.configs;

import org.flywaydb.core.Flyway;
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
import org.springframework.boot.web.servlet.MultipartConfigFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.util.unit.DataSize;

import javax.servlet.MultipartConfigElement;

@Configuration
@ConditionalOnClass(Flyway.class)
@ComponentScan("tv.comnata.comnata")
public class AppConfig {
    @Bean
    MultipartConfigElement multipartConfigElement() {
        MultipartConfigFactory factory = new MultipartConfigFactory();

        factory.setMaxFileSize(DataSize.ofGigabytes(7));
        factory.setMaxRequestSize(DataSize.ofGigabytes(7));

        return factory.createMultipartConfig();
    }
}

File: main-service/src/main/kotlin/tv/comnata/mainservice/configs/AppConfig.kt
package tv.comnata.mainservice.configs

import org.springframework.boot.web.servlet.MultipartConfigFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.util.unit.DataSize
import javax.servlet.MultipartConfigElement

@Configuration
@ComponentScan("tv.comnata.mainservice")
class AppConfig {
    @Bean
    fun multipartConfigElement(): MultipartConfigElement {
        val factory = MultipartConfigFactory()
        factory.setMaxFileSize(DataSize.ofGigabytes(7))
        factory.setMaxRequestSize(DataSize.ofGigabytes(7))
        return factory.createMultipartConfig()
    }
}